presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1321,6 +1324,48 @@ public void testDecimalComparedToWiderDecimal()
                 Range.lessThan(createDecimalType(4, 2), shortDecimal("99.99")), Range.greaterThan(createDecimalType(4, 2), shortDecimal("99.99"))), true));
     }
 
+    @Test
+    public void testVarcharComparedToCharExpression()
+            throws Exception
+    {
+        // greater than or equal
+        testSimpleComparison(greaterThanOrEqual(P, stringLiteral("123456789")), P, Range.greaterThan(createCharType(10), Slices.utf8Slice("12345678")));
```

Comment:
Btw. Is `varchar(x)` coercible to `char(x)`. If not this test does not make sense. Right?
Also do we want to have test for each of such type paris. This is handled by generic mechanism.

Probably instead making this class bloated we should test if `SATURATED_FLOOR_CAST` operator behaves correctly.